### ISSUE TITLE
Experiment with not matching requests -> containers 1:1

### ISSUE THREAD

**Issue by jcrist** [jcrist/skein#136]
Currently a mapping of priority (proxy for a unique request ID) -> pending container is kept to manage all ongoing requests. When a pending container is canceled before starting, its corresponding request is canceled. This is usually fine in real workflows, but sometimes can cause non-ideal behavior. Due to how the YARN allocator works (and what operations are atomic), the following can happen:

- Request a container
- Cancel the request for that container
- Receive the container for the canceled request (at which point we currently release it)

This can lead to situations where a container is received for a canceled request (and thus released), but that container could still fulfill a pending request. Instead of releasing the container, we could instead use it to fulfill a pending request, and cancel the pending request instead.

This rarely happens in real world workflows, but quickly calling:

```
app.scale('myservice', 3)
app.scale('myservice', 1)
```

in a tight loop can cause this behavior. As such this is low priority, but may still be worth thinking about.
